src/components/area/ListTitleArea.js

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react';
+import ModalPortal from '../modal/ModalPortal';
+import InputBox from '../common/InputBox';
+import ShareFolder from '../ShareFolder';
+import shareIcon from '../../assets/icon_share.svg';
+import deleteIcon from '../../assets/icon_delete.svg';
+import editIcon from '../../assets/icon_edit.svg';
+import './ListTitleArea.css';
+
+const ListTitleArea = ({ title }) => {
+  const [isShareModal, setIsShareModal] = useState(false);
+  const [isRenameModal, setIsRenameModal] = useState(false);
+  const [isDeleteFolderModal, setIsDeleteFolderModal] = useState(false);
```

Comment:
모달을 동시에 여러 개 띄우는 게 목적이라면 이렇게 개별의 상태를 관리하는 게 맞겠지만,
모달은 한 번에 하나만 띄우는 게 일반적입니다.

그래서 `'현재 열린 모달이 무엇인지'`에 대한 state를 만들어 관리하시면 됩니다.

```suggestion
  const [currentModal, setCurrentModal] = useState('');
```

그리고 해당 상태에 매핑된 모달을 관리하는 값을 만들고,

```js
const MODAL_MAP = {
  shareModal : <ModalPortal>...</ModalPotal>,
  ...
}
```

실제 사용하는 곳에선 이렇게 사용하시면 됩니다.
```jsx
{MODAL_MAP[currentModal]}
```

그리고 모달을 끄려면, 이런 식으로 할 수 있겠네요.
```js
const closeModal = () => {
  setCurrentModal('');
}
```